dissect/target/loaders/cpio.py

```diff
@@ -0,0 +1,32 @@
+from pathlib import Path
+
+from dissect.util import cpio
+
+from dissect.target import Target
+from dissect.target.filesystems.tar import TarFilesystem
+from dissect.target.helpers.fsutil import open_decompress
+from dissect.target.loader import Loader
+
+
+class CpioLoader(Loader):
+    """Load (compressed) CPIO files."""
+
+    def __init__(self, path: Path, **kwargs):
+        super().__init__(path)
+
+        fh = open_decompress(path, "rb")
+        self.cpio = TarFilesystem(fh, tarinfo=cpio.CpioInfo)
+
+    @staticmethod
+    def detect(path: Path) -> bool:
+        magic = path.open("rb").read(6)
```

Comment:
We might be dealing with real files here, so best to use some context managers for all the opening you're doing in `detect()`.